--- /usr/lib/python3/dist-packages/location/export.py.original	2022-05-09 05:17:31.526837054 +0000
+++ /usr/lib/python3/dist-packages/location/export.py	2022-05-09 05:03:34.147980272 +0000
@@ -47,6 +47,8 @@
 		self.name = name
 		self.family = family
 		self.directory = directory
+		# Tag
+		self.tag = self._make_tag()

 		# Open output file
 		if f:
@@ -58,8 +60,6 @@
 		else:
 			self.f = io.StringIO()

-		# Tag
-		self.tag = self._make_tag()

 		# Call any custom initialization
 		self.init()
