diff --git a/node_modules/next/dist/client/components/hot-reloader.client.js b/node_modules/next/dist/client/components/hot-reloader.client.js
index 77d2b8b..c486623 100644
--- a/node_modules/next/dist/client/components/hot-reloader.client.js
+++ b/node_modules/next/dist/client/components/hot-reloader.client.js
@@ -44,7 +44,12 @@ function HotReload({ assetPrefix  }) {
         if (normalizedAssetPrefix.startsWith('http')) {
             url = `${protocol}://${normalizedAssetPrefix.split('://')[1]}`;
         }
-        webSocketRef.current = new window.WebSocket(`${url}/_next/webpack-hmr`);
+	try {
+            webSocketRef.current = new window.WebSocket(`${url}/_next/webpack-hmr`);
+	} catch (e) {
+	    console.warn("Could not create a WebSocket", e)
+	    webSocketRef.current = null
+	}
     }, [
         assetPrefix
     ]);
diff --git a/node_modules/next/dist/client/dev/error-overlay/websocket.js b/node_modules/next/dist/client/dev/error-overlay/websocket.js
index 4d0acae..f3a59d1 100644
--- a/node_modules/next/dist/client/dev/error-overlay/websocket.js
+++ b/node_modules/next/dist/client/dev/error-overlay/websocket.js
@@ -57,10 +57,15 @@ function connectHMR(options) {
         if (assetPrefix.startsWith('http')) {
             url = `${protocol}://${assetPrefix.split('://')[1]}`;
         }
-        source = new window.WebSocket(`${url}${options.path}`);
-        source.onopen = handleOnline;
-        source.onerror = handleDisconnect;
-        source.onmessage = handleMessage;
+        try {
+            source = new window.WebSocket(`${url}${options.path}`);
+            source.onopen = handleOnline;
+            source.onerror = handleDisconnect;
+            source.onmessage = handleMessage;
+        } catch (e) {
+            console.warn("Cannot create a websocket", e);
+            source = null;
+        }
     }
     init();
 }
